source-mysql/prerequisites.go

```diff
@@ -19,7 +25,14 @@ func (db *mysqlDatabase) SetupPrerequisites(ctx context.Context) []error {
 }
 
 func (db *mysqlDatabase) prerequisiteBinlogFormat(ctx context.Context) error {
-	// TODO: Verify that 'binlog_format' is set to 'ROW'
+	var results, err = db.conn.Execute(`SELECT @@GLOBAL.binlog_format;`)
+	if err != nil || len(results.Values) == 0 {
+		return fmt.Errorf("unable to query 'binlog_format' system variable: %w", err)
+	}
+	var format = string(results.Values[0][0].AsString())
```

Comment:
Is it possible for `results.Values[0]` to be empty?

Reply:
Only in the pedantic sense that we're communicating with a remote server that could in principle be actively malicious and could send us whatever response data it wants.

But practically speaking no, if we get a result row from this query it will contain one value, which is the current setting of the system variable.